url026791d51b_dhleong_SeriesTracker_v6_tgz-pJ8-stv6_SeriesTrackerJ8/src/com/eekboom/utils/Strings.java

```diff
@@ -0,0 +1,353 @@
+     */
+    public static int compareNatural(String s, String t) {
+        return compareNatural(s, t, false, Collator.getInstance());
+    }
+
+    /**
+     * <p>Compares two strings using the given collator and comparing contained numbers based on their numeric
+     * values.</p>
+     *
+     * @param s first string
+     * @param t second string
+     * @return zero iff <code>s</code> and <code>t</code> are equal,
+     *         a value less than zero iff <code>s</code> lexicographically precedes <code>t</code>
+     *         and a value larger than zero iff <code>s</code> lexicographically follows <code>t</code>
+     */
+    public static int compareNatural(Collator collator, String s, String t) {
+        return compareNatural(s, t, true, collator);
+    }
+
+    /**
+     * <p>Compares two strings using each character's Unicode value for non-digit characters and the numeric values off
+     * any contained numbers.</p>
+     * <p>(This will probably make sense only for strings containing 7-bit ascii characters only.)</p>
+     *
+     * @return zero iff <code>s</code> and <code>t</code> are equal,
+     *         a value less than zero iff <code>s</code> lexicographically precedes <code>t</code>
+     *         and a value larger than zero iff <code>s</code> lexicographically follows <code>t</code>
+     */
+    public static int compareNaturalAscii(String s, String t) {
+        return compareNatural(s, t, true, null);
+    }
+
+    /**
+     * <p>Compares two strings using each character's Unicode value - ignoring upper/lower case - for non-digit
+     * characters and the numeric values of any contained numbers.</p>
+     * <p>(This will probably make sense only for strings containing 7-bit ascii characters only.)</p>
+     *
+     * @return zero iff <code>s</code> and <code>t</code> are equal,
+     *         a value less than zero iff <code>s</code> lexicographically precedes <code>t</code>
+     *         and a value larger than zero iff <code>s</code> lexicographically follows <code>t</code>
+     */
+    public static int compareNaturalIgnoreCaseAscii(String s, String t) {
+        return compareNatural(s, t, false, null);
+    }
+
+    /**
+     * @param s             first string
+     * @param t             second string
+     * @param caseSensitive treat characters differing in case only as equal - will be ignored if a collator is given
+     * @param collator      used to compare subwords that aren't numbers - if null, characters will be compared
+     *                      individually based on their Unicode value
+     * @return zero iff <code>s</code> and <code>t</code> are equal,
+     *         a value less than zero iff <code>s</code> lexicographically precedes <code>t</code>
+     *         and a value larger than zero iff <code>s</code> lexicographically follows <code>t</code>
+     */
+    private static int compareNatural(String s, String t, boolean caseSensitive, Collator collator) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 51. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.